### PR TITLE
WIP: Mongo shell authentication

### PR DESCRIPTION
**Pull request by arkilic**
Requires https://github.com/NSLS-II/metadataservice/pull/22

### CODE REVIEW

**Comment by tacaswell** [comment]
Can't we just run this with auth off?  If it is working with auth in metadatastore, metadataserver only talks to mongo via metadatastore and mdsclient only talks to metadata server.

It is probably better to flip the default of `auth` to false in metadatastore and not worry about propogating this change out to everywhere else.

**Comment by arkilic** [comment]
I think the CI  should mimic the production machines. 
metadataservice has 3 additional args: --auth, --mongo-user, and --mongo-pwd.  It passes these args to mdstore, so we should test mdservice with authentication to make sure this mechanism is intact for production use.
We can run mdclient tests w/o authentication as long as you test mdstore and mdservice. I am just being cautious.

File: .travis.yml [hunk 1]
```diff
@@ -18,11 +18,12 @@ before_install:
   - git clone https://github.com/nsls-ii/nsls2-ci ~/scripts
   - . ~/scripts/install-miniconda.sh
 
-  - git clone https://github.com/NSLS-II/metadataservice.git /home/travis/metadataservice
```
Comment (danielballan) [inline .travis.yml]: Is your master branch ahead of the NSLS-II master branch?

Comment (arkilic) [inline .travis.yml]: Yes, as I said above this requires NSLS-II/metadataservice#22

Comment (danielballan) [inline .travis.yml]: Both parts of that -- your master branch being ahead of upstream and putting a user branch in the .travis.yml -- could be messy if we forget to change them back later. Let's get that MDS PR merged into upstream and leave this line as it was. Sound OK?

Comment (arkilic) [inline .travis.yml]: Agreed, that's what we have been doing

Comment (danielballan) [inline .travis.yml]: OK. Please revert this line and then merge at will.

**Pull request merged**